### PR TITLE
adding part 4

### PR DESCRIPTION
**Pull request by amy-u**

### CODE REVIEW

**Comment by vegetabill** [comment]
@amy-u when I tried to test this branch, it didn't work. There was a syntax error on line 161 of index.js. When I removed the broken code, the Ticketmaster search didn't work.

**Comment by vegetabill** [comment]
This isn't working so no need to review.

File: EventonicaPart4/Eventonica/jQuery-scripts.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                        console.log(result);
+                        // Parse the response.
+                        // Do other things.
+                        result._embedded.events.map((anEvent) => {
+                            $("#show-up").append(`<dt> Name: ${anEvent.name} </dt>`);
+                        })
+                     },
+            error: function(xhr, status, err) {
+                        // This time, we do not end up here!
+                     }
```
Comment (datadaveshin) [inline EventonicaPart4/Eventonica/jQuery-scripts.js]: Do something with this?

File: EventonicaPart4/Eventonica/index.html [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        <input type="text" id="keyword-search-id">
+        <br/>
+        <input type="submit">
+    </form>
+    <h4>Results:</h4>
+    <ol id="show-up">
+        <!-- Use jQuery to add search results here-->
+    </ol>
+</div>
+
```
Comment (datadaveshin) [inline EventonicaPart4/Eventonica/index.html]: Maybe add this button here: `A button that says "Import first event from Ticketmaster"`

File: EventonicaPart4/Eventonica/index.js [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+  //     html += `<li>${item.ename} - ${item.description} <ol>${item.searchTickets(0,250)}</ol> </li>`;
+  //   });
+  //   // insert final html into #event...                    
+  //   $("#index").html(html);
+  // });
+
+
+  if (typeOf module != "undefined"){ // double check if I need to delete this line
+    module.exports = { EventRecommender, User, Event}
+  }
```
Comment (datadaveshin) [inline EventonicaPart4/Eventonica/index.js]: I had to comment this part out to get your app to display results. Maybe find a better way to take care of what you were doing here? ;)

File: EventonicaPart4/Eventonica/index.js [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    //   this.availableTickets.push(priceOf); //pushing the new ticket obj into the available ticket array
+    // }
+  
+    allTicket() {
+      let allP = "Available Tickets: "; //store ticket strings
+      $.each(this.availableTickets, function(index, item) {
+        // $.each is like a for loop. It is going thru this.avaiableTickets array and applying a function to each index
+        allP += `<li>${item.tn} ($${item.price})</li>`; //adding more string to the preexisting strings inside of allP
+      }); //item here refers to the object in each index (in this case the objects are tickets. Hense why we use .tn and .price because those are key names in the ticket objects)
+      return allP; //returning the big string
```
Comment (datadaveshin) [inline EventonicaPart4/Eventonica/index.js]: If `allP` was named `availableTicketsString` you wouldn't need the extra comments. 
Try to find some examples from the book Clean Code, like this one [A Brief Guide to Clean Code](https://dev.to/danialmalik/a-beginner-s-guide-to-clean-code-part1-naming-conventions-139l) .  Also the comment on how $.each works and the stuff on line 121 can probably be put in your personal notes. It just makes it harder for your reviewers to sift through your code.

File: EventonicaPart4/Eventonica/index.js [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+    }
+  
+    searchTickets(min,max) {
+      let eT = "Eligible tickets: "; //The eligible ticket will be stored here
```
Comment (datadaveshin) [inline EventonicaPart4/Eventonica/index.js]: I think I commented on this before thinking `eT` is the the extra terrestrial. `eligibleTickets` or `eligibleTicketString` would be easier to figure out what's going on in a larger code base.

File: EventonicaPart4/Eventonica/index.js [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+    //   let priceOf = new TicketType(name, amount); //creating a new tickettype obj into the variable priceOf
+    //   this.availableTickets.push(priceOf); //pushing the new ticket obj into the available ticket array
+    // }
+  
+    allTicket() {
+      let allP = "Available Tickets: "; //store ticket strings
+      $.each(this.availableTickets, function(index, item) {
+        // $.each is like a for loop. It is going thru this.avaiableTickets array and applying a function to each index
+        allP += `<li>${item.tn} ($${item.price})</li>`; //adding more string to the preexisting strings inside of allP
+      }); //item here refers to the object in each index (in this case the objects are tickets. Hense why we use .tn and .price because those are key names in the ticket objects)
```
Comment (datadaveshin) [inline EventonicaPart4/Eventonica/index.js]: Maybe add this comment to your personal notes, and use a more descriptive key than `.tn`?

File: EventonicaPart4/Eventonica/index.js [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+  eventArray.push(/*input of events*/);
+  
+  // $(document).ready(function() {
+  //   let html = "";
+  //   $.each(eventArray, function(index, item) {
+  //     html += `<li>${item.ename} - ${item.description} <ol>${item.searchTickets(0,250)}</ol> </li>`;
+  //   });
+  //   // insert final html into #event...                    
+  //   $("#index").html(html);
+  // });
```
Comment (datadaveshin) [inline EventonicaPart4/Eventonica/index.js]: ?

File: EventonicaPart4/Eventonica/jQuery-scripts.js [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+        if(str == '' && num == ''){
+            alert("Error! Missing name or id nubmer.");
+        }
+        else{$("#all-events").html(`<dt>${str}</dt> <dd><strong>ID:</strong> ${num} <dd>`);
+        }
+    });
+    //Deleting event
+    $("#delete-event").submit((event) => {
+        event.preventDefault();
+        let num = $("#delete-event-id").val();
```
Comment (datadaveshin) [inline EventonicaPart4/Eventonica/jQuery-scripts.js]: Again, something more descriptive than `num` will help in the long run...

**Pull request closed without merging**